blob/blob.go

```diff
@@ -588,19 +601,43 @@ func Open(ctx context.Context, urlstr string) (*Bucket, error) {
 	return NewBucket(drv), nil
 }
 
+// wrappedError is used to wrap all errors returned by drivers so that users
+// are not given access to provider-specific errors.
+type wrappedError struct {
+	err error
+	b   driver.Bucket
+}
+
+func wrapError(b driver.Bucket, err error) error {
+	if err == nil {
+		return nil
+	}
+	if err == io.EOF {
+		return err
+	}
+	if _, ok := err.(*wrappedError); ok {
```

Comment:
May I suggest, instead of trying hard to avoid this, just cut yourself some slack and return `err`. I think that will make the implementation more robust overall.

Reply:
Yea. Done. I updated the tests to check for double-wrapping instead.